src/starknet/starknet_storage.rs

```diff
@@ -128,6 +128,8 @@ impl CommitmentInfo {
     }
 }
 
+// TODO: document this, is this the entire tree storage for a whole contract?
+//       why is it called "Single"?
```

Comment:
TODO: convert this to documentation :)

I'm guessing `Single` mostly comes from the pythonic implementation, but maybe there's a better answer than that.